datasets.py

```diff
@@ -35,5 +35,57 @@ class Dataset:
 
     @classmethod
     def from_text(cls, text: str) -> Self:
-        """Convert a text representing a single dataset into the `Dataset` instance."""
-        raise NotImplementedError
+        """Convert a text representing a single dataset into the `Dataset` instance.
+
+        Args:
+            text (str): String with specific pattern in which each row is separated
+                by "\n"
+
+        Raises:
+            Exception: Input text does not have enough rows to assign values to
+                all attributes of the Dataset class.
+
+        Returns:
+            Self: Returns Dataset class object with data imported from input text.
+        """
+        if text.find("*") != -1:
+            accepted = False
+            text = text.replace("*", "")
+        else:
+            accepted = True
+        text_list = text.split("\n")
+        if len(text_list) < 6:
+            raise Exception(f"Text has too few rows. Expected >6, got {len(text_list)}")
+        elif len(text_list) < 7:
+            print("Warning: No data to import! Check input.")
+        type = (text_list[0].split(":"))[1].strip()
+        if type not in ("density", "viscosity", "surften"):
+            raise ValueError(
+                f"Invalid 'type' value: {type}. "
+                f"Expected one of 'density', 'viscosity', 'surften'."
+            )
+        reference = text_list[1]
+        ionic_liquid = text_list[4]
+        try:
+            water_content_ppm = int((text_list[5].split(";"))[1])
+        except:  # noqa: E722
```

Comment:
Applying `noqa` markers is tempting :smile:, but try to avoid as much as possible.

In general, it is not recommended to use `except` without specifying a type of exception. In this case, I would go with: `TypeError`.

Or... In Python, you can check whether a text is composed of digits (I guess, that is the point here). There is a method `isdigit`:

```python
"12345".isdigit()  # True
"1234s".isdigit()  # False
```